src/alfasim_sdk/alfasim_sdk_api/detail/bootstrap_linux.h

```diff
@@ -44,56 +44,80 @@ inline int alfasim_sdk_open(ALFAsimSDK_API* api)
     // Load shared object
     char *full_filepath = NULL;
     asprintf(&full_filepath, "%s%s", alfasim_executable_dir, SO_FILENAME);
+
+    // https://www.man7.org/linux/man-pages/man3/dlopen.3.html
+    // > If dlopen() fails for any reason, it returns NULL.
     api->handle = dlopen(full_filepath, RTLD_LAZY);
     free(full_filepath);
+    if (api->handle == NULL) {
+        return SDK_FAILED_TO_LOAD_DLL;
+    }
+
+    // https://man7.org/linux/man-pages/man3/dlsym.3.html
+    // > In unusual cases (see NOTES) the value of the symbol could actually be NULL.  Therefore,
+    // > a NULL return from dlsym() need not indicate an error.  The correct way to distinguish an
+    // > error from a symbol whose value is NULL is to call dlerror(3) to clear any old error
+    // > conditions, then call dlsym(), and then call dlerror(3) again, saving its return value
+    // > into a variable, and check whether this saved value is not NULL.
+    #define LOAD_SDK_PROC_MAP_TYPE(func_name, func_type) {\
+        dlerror();\
+        api->func_name = (func_type)dlsym(api->handle, #func_name);\
+        char *last_error = dlerror();
+        if (last_error != NULL) {\
+            dlclose(api->handle);\
+            api->handle = nullptr;\
+            return SDK_FAILED_TO_LOAD_FUNCTION;\
+        }\
+    }
+    #define LOAD_SDK_PROC(func_name) LOAD_SDK_PROC_MAP_TYPE(func_name, func_name ## _func)
```

Comment:
It is many years since I wrote any macro.
Is this (and the ones in the other bootstrap file) OK?